internals/approval_defs.py

```diff
@@ -206,27 +208,39 @@ def get_gate_by_type(feature_id: int, gate_type: int):
     return None
   return gates[0]
 
-def _calc_gate_status(gate: Gate) -> int:
-  """Evaluates the current state that this gate should have."""
-  votes = Vote.get_votes(gate_id=gate.key.integer_id())
-  approvals = 0
-  for vote in votes:
-    if vote.state in (Vote.APPROVED, Vote.NA):
-      approvals += 1
-    elif vote.state == Vote.DENIED:
-      return Vote.DENIED
-  afd = APPROVAL_FIELDS_BY_ID[gate.gate_type]
 
-  if ((afd.rule == ONE_LGTM and approvals >= 1) or
-      (afd.rule == THREE_LGTM and approvals >= 3)):
-    return Vote.APPROVED
-  return gate.state
+def _calc_gate_state(votes: list[Vote], rule: str) -> int:
+  """Returns the state that a gate should have based on its votes."""
```

Comment:
Thanks for taking a look at this. The old implementation would not revert to a non-approved state if one of the API owners approved but then set their state back to undecided, so my plan was to revisit this.